### ISSUE TITLE
Splitting left and right AABB subtree

### ISSUE THREAD

**Issue by koichishi** [dilevin/computer-graphics-bounding-volume-hierarchy#44]
Do we have to split a box into roughly equal-spaced subtrees?
When the left tree contains only one object and the right tree has 10000000000 objects, can we separate the right tree again and move half of the nodes to the left tree?

**Comment by chl9797** [comment]
No. One object in the left tree and the rest in the right tree is fine.
Note that in splitting, there's a failure mode where a single long triangle determines the bounds of the entire box and everything goes to one side. An easy way to deal with this is to pull one random object from one side and put it into the other.
